src/test/shared/ui/picker.test.ts

```diff
@@ -366,3 +367,334 @@ describe('promptUser', async () => {
+    })
+
+    it('returns undefined if no override is provided', async () => {
+        const spy = sandbox.spy()
+        const out = await picker.iteratingOnDidTriggerButton(
+            { iconPath: new vscode.ThemeIcon('squirrel') },
+            () => {},
+            () => {},
+            new fakeIteratingQuickPickController(spy)
+        )
+        assert.strictEqual(out, undefined)
+        assert.ok(spy.notCalled)
+    })
+
+    it('returns a value from the override function', async () => {
+        const spy = sandbox.spy()
+        const items: vscode.QuickPickItem[] = [{ label: 'asdf' }, { label: 'jkl;' }]
+        const out = await picker.iteratingOnDidTriggerButton(
+            { iconPath: new vscode.ThemeIcon('squirrel') },
+            () => {},
+            () => {},
+            new fakeIteratingQuickPickController(spy),
+            async () => {
+                return items
+            }
+        )
+        assert.deepStrictEqual(out, items)
+        assert.ok(spy.notCalled)
+    })
+})
+
+describe('IteratingQuickPickPopulator', async () => {
+    it('transforms values from the iterator and can reset itself', async () => {
+        const values = ['a', 'b', 'c']
+        async function* iteratorFn(): AsyncIterator<string> {
+            for (const val of values) {
+                yield val
+            }
+        }
+        const populator = new picker.IteratingQuickPickPopulator<string>(
+            () => iteratorFn(),
+            val => {
+                if (val) {
+                    return [{ label: val.toUpperCase() }]
+                }
+
+                return []
+            }
+        )
+
+        const firstIter = populator.getPickIterator()
+        let firstI = 0
+        let firstItem = await firstIter.next()
+        while (!firstItem.done) {
+            assert.ok(Array.isArray(firstItem.value)),
+                assert.strictEqual(firstItem.value.length, 1),
+                assert.deepStrictEqual(firstItem.value[0], { label: values[firstI].toUpperCase() })
+            firstI++
+            firstItem = await firstIter.next()
+        }
+
+        populator.reset()
+
+        const secondIter = populator.getPickIterator()
+        let secondI = 0
+        let secondItem = await secondIter.next()
+        while (!secondItem.done) {
+            assert.ok(Array.isArray(secondItem.value)),
+                assert.strictEqual(secondItem.value.length, 1),
+                assert.deepStrictEqual(secondItem.value[0], { label: values[secondI].toUpperCase() })
+            secondI++
+            secondItem = await secondIter.next()
+        }
+    })
+})
+
+describe('IteratingQuickPickController', async () => {
+    const values = ['a', 'b', 'c']
+    const result = [{ label: 'A' }, { label: 'B' }, { label: 'C' }]
+    const errMessage = 'ahhhhhhhhh!!!'
+
+    async function* iteratorFn(): AsyncIterator<string> {
+        for (const [i, value] of values.entries()) {
+            await new Promise(resolve => {
```

Comment:
Please have a look at some existing tests that use lolex to control time based tests. It shouldn't be acceptable to have tests that run up the time needed for the test suite as a whole - try to aim for each test taking no more than 100ms.

Reply:
As a side note, since you brought it up, lolex is now renamed and the lolex package deprecated, we should migrate it in another PR https://www.npmjs.com/package/lolex

Reply:
Yup, will do in a fast-follow PR. Don't want to touch other tests unrelated to this change here.

This is also why I couldn't use lolex 6.0.0; they only had types through 5.1.0!